doc/developer-notes.md

```diff
@@ -1080,6 +1080,10 @@ A few guidelines for introducing and reviewing new RPC interfaces:
     until the wallet is caught up to the chainstate as of the RPC call's entry.
     This also makes the API much easier for RPC clients to reason about.
 
+- The `getbalances` RPC now returns `lastblock` which contains the wallet's last processed block
```

Comment:
This seems to be information for the release note?

Reply:
🤣 indeed! my brain read release-notes.md..

Reply:
Should I change anything? Add further info?

Reply:
Move this hunk to release notes, not developer notes.

Reply:
Ok, but what about the rest of the paragraph in developer-notes? Maybe I should rather keep it there as well?

Reply:
@promag It seems, I can only change the 0.20.0 release (draft) notes here: https://github.com/bitcoin-core/bitcoin-devwiki/wiki/0.20.0-Release-Notes-Draft

The release-noted.md I have is pointing at it. 

I'm not sure if I should put an entry in 0.20.0 release notes when this PR is still unmerged.

Reply:
@brakmic I don't know if this is for 0.20, as the feature freeze was 3 weeks ago, but see the release notes section in developer-notes.md. Normally you don't edit release-notes.md, rather instead you create a file in doc/release-notes/ called release-note-18570.md.

Reply:
Added respective doc.